### PR TITLE
[Bug] Fix crash when using enemy move override

### PR DESCRIPTION
**Pull request by DayKev**
## What are the changes the user will see?
Potentially some missing animation issues will no longer cause a crash... or it's possible the issue is only caused specifically by the override.

## Why am I making these changes?
It's really annoying when the game crashes while testing. 

## What are the changes from a developer perspective?
The game will crash less when using the enemy moveset override.

### Screenshots/Videos
Before fix:
image

## How to test the changes?
```ts
const overrides = {
  OPP_MOVESET_OVERRIDE: Moves.SPLASH
} satisfies Partial<InstanceType<typeof DefaultOverrides>>;
```
Start a new Classic run with any random Pokémon, make sure not to faint the enemy on the first turn.

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- ~[ ] Have I considered writing automated tests for the issue?~
- ~[ ] If I have text, did I make it translatable and add a key in the English locale file(s)?~
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- ~[ ] Are the changes visual?~
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/data/battle-anims.ts [hunk 1]
```diff
@@ -1279,7 +1279,7 @@ export class MoveAnim extends BattleAnim {
   getAnim(): AnimConfig {
     return moveAnims.get(this.move) instanceof AnimConfig
       ? moveAnims.get(this.move) as AnimConfig
-      : moveAnims.get(this.move)![this.user?.isPlayer() ? 0 : 1] as AnimConfig; // TODO: is this bang correct?
+      : moveAnims.get(this.move)?.[this.user?.isPlayer() ? 0 : 1] as AnimConfig;
```
Comment (innerthunder) [inline src/data/battle-anims.ts:1282]: Why the `.` here?

**Review by ben-lear — APPROVED** [review]

File: src/data/battle-anims.ts (hunk 1 above)
Comment (ben-lear) [reply src/data/battle-anims.ts:1282]: Optional chaining requires it, even if you’re about to index an areay

Comment (DayKev) [reply src/data/battle-anims.ts:1282]: That's the syntax for using a `?` before `[]`, you have to add the `.` for whatever reason. 🤷
I asked the same question recently lmao.

Comment (flx-sta) [reply src/data/battle-anims.ts:1282]: this is necessary when using the ternary

**Review by flx-sta — APPROVED** [review]
it fixes it, but now the enemy pokemon doesn't perform any animations anymore (which is obviously the cause, but why?)

Only when using overrides!

**Review by torranx — APPROVED** [review]
lgtm

**Pull request merged**